### PR TITLE
Added Swagger and Vendor controller (Elizaveta Dubogryz)

### PR DESCRIPTION
**Pull request by el-dub**

### CODE REVIEW

**Review by ghost — DISMISSED** [review]
Changes were agreed upon between me and the given programmer.

**Review by ghost — DISMISSED** [review]
Ювелирная работа

File: src/main/java/com/exadel/discount/controller/VendorController.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            ), "Description")
+    );
+
+    private final List<BaseVendorDTO> baseVendorsServiceMock = List.of(
+            new BaseVendorDTO(2, "Vendor2")
+    );
+
+    @GetMapping
+    @ApiOperation("Get list of all vendors")
+    public ResponseEntity<List<BaseVendorDTO>> getVendorsList() {
```
Comment (SergeyKoval) [inline src/main/java/com/exadel/discount/controller/VendorController.java]: You need ResponseEntity only if you want to set status code dynamically (or something similar)

File: src/main/java/com/exadel/discount/controller/VendorController.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        return ResponseEntity
+                .status(HttpStatus.OK)
+                .body(baseVendorsServiceMock);
+    }
+
+    @GetMapping("/{vendorId}")
+    @ApiOperation("Get vendor by ID")
+    public ResponseEntity<DetailedVendorDTO> getVendorById(@PathVariable(name = "vendorId") long id) {
```
Comment (SergeyKoval) [inline src/main/java/com/exadel/discount/controller/VendorController.java]: Introduce spring validation. vendorId is required here

**Review by SergeyKoval — CHANGES_REQUESTED** [review]

File: src/main/java/com/exadel/discount/controller/VendorController.java (hunk 2 above)
Comment (SergeyKoval) [reply src/main/java/com/exadel/discount/controller/VendorController.java]: Or you can leave as it is and create separate task to introduce spring validation

**Review by ghost — APPROVED** [review]
I do not see any errors, so this commit has a right to be on the origin branch

File: src/main/java/com/exadel/discount/controller/VendorController.java (hunk 1 above)
Comment (el-dub) [reply src/main/java/com/exadel/discount/controller/VendorController.java]: +

File: src/main/java/com/exadel/discount/controller/VendorController.java (hunk 2 above)
Comment (el-dub) [reply src/main/java/com/exadel/discount/controller/VendorController.java]: +

**Review by SergeyKoval — APPROVED** [review]

**Pull request merged**